include/copy_board.h

```diff
@@ -1,7 +1,7 @@
 #ifndef COPY_BOARD_H_INCLUDED
 #define COPY_BOARD_H_INCLUDED
 
-void copy_board(int rows, int cols, int *board, int *copy);
+void copy_board(int rows, int cols, int board[rows][cols], int copy[rows][cols]);

```

Comment:
Does it even work?

Reply:
Spoiler: no

Reply:
It does actually.
Point taken though, I did not intend to keep using this notation.